### PR TITLE
Initialise return values to an error

### PR DESCRIPTION
**Pull request by yanesca**
Initialising the return values to and error is best practice and makes the library more robust.
    
No new error was introduced because we are running out of error codes and we shouldn't be wasting them. 

We already had an error code reserved for generic errors. This PR makes it explicit and uses this error code to initialise return values.

### CODE REVIEW

**Comment by yanesca** [comment]
CI fails because Mbed OS uses Mbed TLS's `error.h` and should pass once Mbed TLS has been updated.

File: library/psa_crypto.c [hunk 1]
```diff
@@ -351,6 +351,14 @@ static psa_status_t mbedtls_to_psa_error( int ret )
             return( PSA_ERROR_INSUFFICIENT_MEMORY );
         case MBEDTLS_ERR_ECP_HW_ACCEL_FAILED:
             return( PSA_ERROR_HARDWARE_FAILURE );
+        /*
+         * Mbed Crypto return values are initialised to
+         * MBEDTLS_ERR_GENERIC_ERROR.  PSA return valuse are initialised to
+         * PSA_ERROR_CORRUPTION_DETECTED.  We are mapping these to each other
+         * to align this behaviour.
```
[the remark this answers is not in the document]
Comment (yanesca) [reply library/psa_crypto.c]: Ok, I have added a new error code.

**Comment by yanesca** [comment]
I have added a new error code for this purpose and simplified the approach as well. The PR is ready for review again.

**Comment by gilles-peskine-arm** [comment]
There are many CI failures. I don't think this PR is ready.

**Comment by yanesca** [comment]
CI fails because Mbed OS uses Mbed TLS's error.h and should pass once Mbed TLS has been updated.

**Comment by yanesca** [comment]
In retrospect I probably should have started with the Mbed TLS PR, but now I would prefer to continue with the review of this one and raise the Mbed TLS counterpart when this is approved. (I would like to do them sequentially to minimise duplicating work during review.)

**Comment by gilles-peskine-arm** [comment]
Ok. Reviewing.

File: include/mbedtls/error.h [hunk 2]
```diff
@@ -55,6 +55,7 @@
  * Low-level module errors (0x0002-0x007E, 0x0003-0x007F)
```
Comment (AndrzejKurek) [inline include/mbedtls/error.h]: Please update this information.

File: include/mbedtls/error.h [hunk 3]
```diff
@@ -52,9 +52,10 @@
  * For historical reasons, low-level error codes are divided in even and odd,
  * even codes were assigned first, and -1 is reserved for other errors.
  *
- * Low-level module errors (0x0002-0x007E, 0x0003-0x007F)
+ * Low-level module errors (0x0001-0x007E, 0x0003-0x007F)
```
Comment (AndrzejKurek) [inline include/mbedtls/error.h]: Shouldn't it rather be (even, odd)?
```suggestion
 * Low-level module errors (0x0002-0x007E, 0x0001-0x007F)
```

Comment (yanesca) [reply include/mbedtls/error.h]: Indeed, my bad.

**Pull request merged**